.github/workflows/reusable_docker_pipeline.yml

```diff
@@ -240,4 +260,43 @@ jobs:
       
       - name: Inspect image
         run: |
-          docker buildx imagetools inspect ${{ vars.AWS_ECR_REGISTRY_ID }}/${{ needs.prepare-metadata.outputs.image-name }}:${{ steps.meta.outputs.version }}
\ No newline at end of file
+          docker buildx imagetools inspect ${{ vars.AWS_ECR_REGISTRY_ID }}/${{ needs.prepare-metadata.outputs.image-name }}:${{ steps.meta.outputs.version }}
+
+  vulnerability_scan:
+    runs-on: ubuntu-24.04
+    if: inputs.publish
+    needs:
+    - merge_dockerhub
+    - prepare-metadata
+    steps:
+    - name: Create Trivy config
+      run: |
+        cat <<EOF > /tmp/trivy.yaml
+        format: sarif
+        exit-code: '0'
+        output: trivy-results.sarif
+        vulnerability:
+          ignore-unfixed: true  
+          type: 
+            - os
+            - library
+        severity: 
+          - CRITICAL
+          - HIGH
+        db:
+          repository: index.docker.io/aquasec/trivy-db:2
```

Comment:
Would it help against rate-limiting to upload this image to our dockerhub and use docker auth?

Reply:
Yes, it will definitely helps! But we will also need a cronjob to update the database regularly because Trivy updates their database every 3 hours.
Since i switch to dockerhub repo instead of the default ghcr, I haven't hit any rate-limit yet. As i see, dockerhub rate limit is 100 pulls/6hours per IP address(runner ip address maybe), if we use below that rate limit, maybe we can skip the database cloning?

Reply:
As per our offline sync, we can include docker auth (which we currently use only for the docker publish job) if we start hitting docker limits.

Reply:
> As per our offline sync, we can include docker auth (which we currently use only for the docker publish job) if we start hitting docker limits.

I just found out that Trivy use the 2 environment: TRIVY_USERNAME and TRIVY_PASSWORD to auth docker (which we already do) 
Referal: [Trivy Database](https://aquasecurity.github.io/trivy/v0.40/docs/vulnerability/db/), under "Private Hosting" section.